src/service/viewer-impl.js

```diff
@@ -276,10 +276,17 @@ export class Viewer {
      */
     this.messagingReadyPromise_ = timer.timeoutPromise(
         20000,
-        new Promise(resolve => {
+        new Promise((resolve, reject) => {
           /** @private @const {function(!Viewer)} */
           this.messagingReadyResolver_ = resolve;
+
+          if (!this.isEmbedded_) {
+            reject(new Error('Page is not embedded.'));
+          }
         })).catch(reason => {
+          if (!this.isEmbedded_) {
+            return;
```

Comment:
Shouldn't we say:

if (!this.isEmbedded) {
this.messagingReadyPromise_ = Promise.reject(new Error('not meeded'))
}

and then down where we postMessage just not do it?